src/components/ListItems/index.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import {
+  DeleteButton,
+  IconCheck,
+  IconRemove,
+  List,
+  StyledItem,
+} from "./style";
+import { Task } from "../../services/taskService";
+
+interface ListItemsProps {
+  items: Task[];
+  completedItems: Task[];
+  pendingItems: Task[];
+  onDeleteItem: (id: number) => void;
+  onCheckClick: (id: number) => void;
+  selectedButton: string | null;
+  searchTerm: string;
+}
+
+function ListItems({
+  items,
+  completedItems,
+  pendingItems,
+  onDeleteItem,
+  onCheckClick,
+  selectedButton,
+  searchTerm,
+}: ListItemsProps) {
+  const [activeItem, setActiveItem] = useState<number | null>(null);
+
+  const handleItemClick = (index: number) => {
+    setActiveItem(index === activeItem ? null : index);
+  };
+
+  const filterAndMapItems = (items: Task[]) => {
+    return items
+      .filter((item) =>
```

Comment:
In this case we can externalize this filter into a variable so that we do not have a chain of HOFs linked together